idea-plugin/src/main/java/com/palantir/gradle/jdks/PluginUpdateCheckerService.java

```diff
@@ -0,0 +1,77 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.intellij.externalDependencies.DependencyOnPlugin;
+import com.intellij.externalDependencies.ExternalDependenciesManager;
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.PluginManagerCore;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationGroupManager;
+import com.intellij.notification.NotificationType;
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.extensions.PluginId;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.updateSettings.impl.pluginsAdvertisement.PluginsAdvertiser;
+import com.intellij.util.text.VersionComparatorUtil;
+import java.util.Optional;
+import java.util.Set;
+
+@Service(Service.Level.PROJECT)
+public final class PluginUpdateCheckerService {
+
+    private static final String PLUGIN_ID = "palantir-gradle-jdks";
+    private final Logger logger = Logger.getInstance(PluginUpdateCheckerService.class);
+    private final Project project;
+
+    public PluginUpdateCheckerService(Project project) {
+        this.project = project;
+    }
+
+    public void checkPluginVersion() {
+        PluginId pluginId = PluginId.getId(PLUGIN_ID);
+        IdeaPluginDescriptor pluginDescriptor = PluginManagerCore.getPlugin(pluginId);
+        if (pluginDescriptor == null) {
+            logger.info("Plugin " + pluginId + " not found");
+            return;
+        }
+        Optional<String> maybeMinVersion =
+                ExternalDependenciesManager.getInstance(project).getDependencies(DependencyOnPlugin.class).stream()
+                        .filter(dependencyOnPlugin ->
+                                dependencyOnPlugin.getPluginId().equals(pluginId.getIdString()))
+                        .map(DependencyOnPlugin::getMinVersion)
+                        .filter(version -> Optional.ofNullable(version).isPresent())
+                        .max(VersionComparatorUtil::compare);
+        boolean isPluginUpToDate = maybeMinVersion
+                .map(minVersion -> VersionComparatorUtil.compare(pluginDescriptor.getVersion(), minVersion) > 0)
+                .orElse(true);
+
+        if (!isPluginUpToDate) {
+            Notification notification = NotificationGroupManager.getInstance()
+                    .getNotificationGroup("Update Palantir plugins")
+                    .createNotification(
+                            "Update palantir-gradle-jdks plugin",
+                            String.format(
+                                    "Please update the plugin in the Settings window to a version higher than '%s'",
+                                    maybeMinVersion.get()),
+                            NotificationType.ERROR);
+            notification.notify(project);
```

Comment:
From the video it looks like if you click the notification it takes you to the update page. But I don't think that's super obvious it's going to do that. Can we add a link or button like the install extra plugins button?

<img width="332" alt="Screenshot 2024-09-12 at 12 56 23" src="https://github.com/user-attachments/assets/86c92b58-2591-40cc-926b-a4ea9016b314">

Reply:
> From the video it looks like if you click the notification it takes you to the update page. 

It opens up both the notification and the window to install the plugins. In the video I was just trying to highlight that there is also a pop-up that shows up, but I didn't click on it. 

Do you think it is better to just have the pop-up with an "Update" button and if the user clicks on it to actually open the Installation window ?